src/frontend/components/property-type/currency/edit.tsx

```diff
@@ -0,0 +1,38 @@
+import { CurrencyInput, CurrencyInputProps, FormGroup, FormMessage } from '@adminjs/design-system'
+import React, { FC, memo, useEffect, useState } from 'react'
+import { EditPropertyProps } from '../base-property-props'
+import { recordPropertyIsEqual } from '../record-property-is-equal'
+import { PropertyLabel } from '../utils/property-label'
+
+type CurrencyEditPropertyProps = EditPropertyProps & CurrencyInputProps
+
+const Edit: FC<CurrencyEditPropertyProps> = (props) => {
+  const { onChange, property, record } = props
+  const propValue = record.params?.[property.path] ?? ''
+  const [value, setValue] = useState(propValue)
+  const error = record.errors?.[property.path]
+
+  useEffect(() => {
+    if (value !== propValue) {
+      setValue(propValue)
+    }
+  }, [propValue])
```

Comment:
It shouldn't be done like this - it triggers unnecessary rerenders. Instead, split it into two components, keep `propValue` in the outer one, use it as a `key` for the inner one, and move the `value` state to the inner component. This way if the `key` changes, React will create a new instance of that component with fresh state, all within a single rendering pass.